File: TestProblem/TestProblem.py
class Base:
	def sum(self, a, b):
		return a + b

class TestProblem(Base):
	def sum(self, a, b):
		try:
			with open('/etc/passwd', 'r') as f:
				print('Open /etc/passwd is allowed')
				close(f)
		except Exception as e:
			print('Open /etc/passwd is dis-allowed: ', e)
        # super() is new in Python3
		return super().sum(a, b)

File: TestProblem/TestProblem.cc
#include <fcntl.h>
#include <unistd.h>
#include <iostream>
using namespace std;

template<typename... Args>
auto tsum(Args... args) {
  // This (args + ...) expression is new in c++ 17
  return (args + ...);
}

class TestProblem {
  public:
    int sum(int a, int b) {
      int f = open("/etc/passwd", O_RDONLY);
      if (f == -1) {
        cout << "Open /etc/passwd is dis-allowed";
      } else {
        cout << "Open /etc/passwd is allowed";
        close(f);
      }
      return tsum(a, b);
    } 
};